### ISSUE TITLE
Seed ratio is set to unlimited for new torrents sent to Transmission

### ISSUE THREAD

**Issue by reconman** [pymedusa/Medusa#9331]
**Describe the bug**
Since around 3 weeks ago, new torrents added to Transmission don't have Transmission's default seed ratio of 8.0 set anymore, instead the seed ratio is set to unlimited. In the past, this worked without any problems.

I think this is because of a change introduced in #8485, but I don't know what exactly caused it.

**To Reproduce**
1. Set up Transmission as client
2. Configure a seed ratio in the Transmission settings
3. Let Medusa add a torrent
4. The seed ratio is unlimited

**Expected behavior**
The seed ratio should be the default one from Transmission.

**Medusa (please complete the following information):**
- OS: Docker container
- Branch: master
- Commit: 55cecb9e821fb15f20efe75b0f59d13e896e1d87
- Python version: Unknown
- Database version: Unknown

**Debug logs (at least 50 lines):**
General > Advanced Settings > Enable debug
<details>

```
2021-03-09 18:37:08 DEBUG	SNATCHQUEUE-SNATCH-352408 :: [55cecb9] Transmission: Response to POST request is {"arguments":{},"result":"success"}
2021-03-09 18:37:08 DEBUG	SNATCHQUEUE-SNATCH-352408 :: [55cecb9] POST URL: http://192.168.0.2:9091/transmission/rpc/rpc [Status: 200]
2021-03-09 18:37:08 DEBUG	SNATCHQUEUE-SNATCH-352408 :: [55cecb9] Transmission: Requested a POST connection to http://192.168.0.2:9091/transmission/rpc/rpc with params: None Data: {"arguments": {"ids": ["6c64a7b30bf34ce87d7f0fdfb03f119086ee3bad"], "priority-high": [], "queuePosi... Files: None
2021-03-09 18:37:08 DEBUG	SNATCHQUEUE-SNATCH-352408 :: [55cecb9] Transmission: Response to POST request is {"arguments":{},"result":"success"}
2021-03-09 18:37:08 DEBUG	SNATCHQUEUE-SNATCH-352408 :: [55cecb9] POST URL: http://192.168.0.2:9091/transmission/rpc/rpc [Status: 200]
2021-03-09 18:37:08 DEBUG	SNATCHQUEUE-SNATCH-352408 :: [55cecb9] Transmission: Requested a POST connection to http://192.168.0.2:9091/transmission/rpc/rpc with params: None Data: {"arguments": {"ids": ["6c64a7b30bf34ce87d7f0fdfb03f119086ee3bad"], "seedRatioLimit": 0, "seedRatio... Files: None
2021-03-09 18:37:08 DEBUG	SNATCHQUEUE-SNATCH-352408 :: [55cecb9] Transmission: Response to POST request is {"arguments":{"torrent-added":{"hashString":"6c64a7b30bf34ce87d7f0fdfb03f119086ee3bad","id":92,"name":"[DKB] Tensei shitara Slime Datta Ken 2nd Season - S02E09 [1080p][HEVC-265 10bit][Multi-Subs].mkv"}},"result":"success"}
2021-03-09 18:37:08 DEBUG	SNATCHQUEUE-SNATCH-352408 :: [55cecb9] POST URL: http://192.168.0.2:9091/transmission/rpc/rpc [Status: 200]
```
</details>

**Comment by reconman** [comment]
By debugging I found out that the Provider Options for my Search Providers had the default value of `-1`. `-1` gets changed to `0` in

https://github.com/pymedusa/Medusa/blob/25b080ec12c328d176410fe3fe056deae4c1c0a6/medusa/clients/torrent/transmission.py#L138

I had to change the seed ratio to 8.0 for all of my search providers one by one. I'm pretty sure the logic was different in the past.

**Comment by p0psicles** [comment]
The ratio is purely to run an action in medusa. Its not supposed to set a ratio in transmission. Also it never has done that.

**Comment by reconman** [comment]
> The ratio is purely to run an action in medusa

It's logging a POST request to transmission containing the seedRatio. If I set the seed ratio to `7` in the provider options, I see the seed ratio set to `7` in transmission. You're definitely wrong.

**Comment by p0psicles** [comment]
I think that's some legacy behavior. I'll remove that

**Comment by p0psicles** [comment]
I did an attempt at fixes this. Could you try out branch `feature/fix-seed-ratio`

**Comment by reconman** [comment]
Works!

On another note, I wasn't sure if it's just my imagination, but now I know for sure.

When you change the provider seed ratio and then restart the Docker container, the seed ratio is reset to `-1` again. Can probably also be reproduced outside of docker containers by executing a `kill` (SIGTERM) on Medusa and then starting it again. I think there's some issue with the config persistence.

**Comment by joydeepsaha05** [comment]
I have this issue as well, the seed ratio resets to -1 for the search providers. Using 0.5.10, haven't been able to test out the changes made in the other branch mentioned.

**Comment by tmrbaset** [comment]
I have the 2 above mentioned issues as well, seed ratio is sent to transmission as unlimited, and provider ratio resets to -1 when Medusa restarts. This really hurt my internet quota.

The proposed branch above is now deleted, is it planned to be merged with master branch soon? and does it fix both issues?

**Comment by p0psicles** [comment]
It's in develop branch. 
Not long until release

**Comment by reconman** [comment]
@p0psicles Was the issue of the providerSeedRatios resetting after restart also addressed in develop?

**Comment by p0psicles** [comment]
Haven't noticed that

**Comment by reconman** [comment]
I think I found the cause.

https://github.com/pymedusa/Medusa/blob/develop/medusa/providers/generic_provider.py#L891 is probably used to write the configuration to JSON. When you restart pymedusa, the settings are loaded from the JSON file.

https://github.com/pymedusa/Medusa/blob/develop/medusa/providers/torrent/torrent_provider.py introduces some attributes which aren't included in the to_json method, which are `ratio`, `minseed`, `minleech`, and `client_ratio`.

In order to keep those config values after restart, they need to be written to the JSON string.

If you want, I can open another issue for this.

**Comment by reconman** [comment]
I can even go further and fix the problem with a PR. I think I know how to fix it properly, not 100 % sure. Do you know where the JSON files are turned into Provider objects again?

**Comment by p0psicles** [comment]
@reconman The issue that providers seed ratio is not stored over restart should be fixed with #9413 
Can you confirm?

**Comment by reconman** [comment]
Works! Thank you very much.
